### PR TITLE
Remove posix interop 

### LINKED ISSUE #324: Consider getting rid of common locks and posix interop
Currently, atomicfu provides access to two [on first glance] distinct API sets that are off overall atomicfu direction and are rather legacy of the past that obstruct the reasoning, maintenance and performance work of other libraries:

1) Interoperability API. https://github.com/Kotlin/kotlinx-atomicfu/blob/master/atomicfu/src/nativeInterop/cinterop/interop.def

This one leaks to public API (due to how interop is implemented in K/N) and creates additional difficulties in train/aggregate/overall maintenance. This interop part can be completely removed and replaced with the corresponding calls from `posix.platform` package (e.g. `platform.posix.pthread_mutex_init()`) .
We consider this step fully backwards compatible.

2) `kotlinx.atomicfu.locks` package. 
With the rising popularity of multiplatform, `atomicfu` starts to be [transitively] present in various projects, including other foundational libraries ([link](https://grep.app/search?q=kotlinx.atomicfu.locks.synchronizedobject&filter[lang][0]=Kotlin), e.g. Ktor), meaning that our quite ad-hoc and tailored for a **very** specific need API (`SyncrhonizedObject`) is leaking into public surface.
Effectively it means that soon we won't be able to do anything with `kotlinx.atomicfu.locks` package, sealing its public API.

I propose to start decommissioning it -- starting from warning-level deprecation, eventually raising it to error; in `kotlinx.coroutines` we can just roll our own, tailored `SyncrhonizedObject`/`ReentrantLock`

### PR DESCRIPTION
**Pull request by mvicsokolova**
Fixes #324

The empty `nativeInterop/cinterop/interop.def` is left for compatibility, the corresponding issue: [KT-68411](https://youtrack.jetbrains.com/issue/KT-68411#focus=Comments-27-9848407.0-0),  [KT-60874 ](https://youtrack.jetbrains.com/issue/KT-60874/KLIB-Resolve-Dont-save-transitive-dependencies-in-manifests-depends-property)

### CODE REVIEW

File: atomicfu/build.gradle [hunk 1]
```diff
@@ -171,10 +208,12 @@ if (rootProject.ext.native_targets_enabled) {
             }
         }
 
-        configure(nativeCompilations) {
```
Comment (dkrasnoff) [inline atomicfu/build.gradle:174]: We can remove `nativeCompilations` from this file at all.

File: atomicfu/build.gradle [hunk 2]
```diff
@@ -171,10 +208,12 @@ if (rootProject.ext.native_targets_enabled) {
...
-            cinterops {
-                interop {
-                    defFile 'src/nativeInterop/cinterop/interop.def'
+        targets.withType(KotlinNativeTarget).configureEach { nativeTarget ->
```
Comment (dkrasnoff) [inline atomicfu/build.gradle:211]: Could you please leave a comment in the code about why we need this block? As I understand from this PR description, we can remove it when [KT-60874](https://youtrack.jetbrains.com/issue/KT-60874/KLIB-Resolve-Dont-save-transitive-dependencies-in-manifests-depends-property) is done

**Review by dkrasnoff — APPROVED** [review]

**Comment by dkrasnoff** [comment]
All the comment above are nits and optional to fix.

Good job, thanks 👍

File: atomicfu/build.gradle [hunk 3]
```diff
@@ -126,42 +128,77 @@ kotlin {
...
 
             nativeTest {}
 
+            nativeUnixLikeMain {
+                kotlin.srcDir("src/nativeUnixLike/kotlin")
+                dependsOn nativeMain
+            }
+
+            androidNative32BitMain {
+                kotlin.srcDir("src/androidNative32Bit/kotlin")
```
[the remark this answers is not in the document]
Comment (mvicsokolova) [reply atomicfu/build.gradle:191]: Fixed that for nativeUnixLike as well)

**Comment by mvicsokolova** [comment]
Fixed all the notes and rebased on the fresh develop

File: atomicfu/build.gradle [hunk 4]
```diff
@@ -126,42 +128,77 @@ kotlin {
...
+        androidNativeX86()
+        androidNativeX64()
+        mingwX64()
+        watchosDeviceArm64()
+
+        applyDefaultHierarchyTemplate { target ->
+            target.group("native") { nativeTargets ->
+                nativeTargets.group("nativeUnixLike") {
+                    it.withLinux()
+                    it.withIos()
```
[the remark this answers is not in the document]
Comment (dkrasnoff) [reply atomicfu/build.gradle:160]: sorry for the late message, for some reason it was left pending and I've noticed it only now

**Pull request merged**